activiti-engine/src/test/java/org/activiti/engine/test/api/variables/TransientVariablesTest.java

```diff
@@ -51,6 +51,12 @@ public void testSetTransientVariableInServiceTask() {
     assertNull(runtimeService.getVariable(processInstance.getId(), "response"));
   }
   
+  // Test case for ACT-4304
+  @Deployment
+  public void testTransientVariableDeletion() {
+    runtimeService.startProcessInstanceByKey("transientVarsTest");
```

Comment:
adding an assert here might stop Codacy from complaining

Reply:
@balsarori 
I fixed it. Please review again.
Also, should I backport this fix to 6.x branch that you first fixed?

Reply:
@daisuke-yoshimoto yes please will be great to port it to 6.x